### ISSUE TITLE
Rewrite setup.py / requirements

### ISSUE THREAD

**Issue by ralsina** [rst2pdf/rst2pdf#738]
It's currently very old fashioned (as is to be expected ;-)

It should be redone using something like [poetry](https://github.com/sdispater/poetry) after the py3 port

**Comment by kyluca** [comment]
Hi,

Just wanting to check if this is still desirable to the project? If so, I'd be happy to put up a PR for Hacktoberfest?

Poetry was a great suggestion in 2019, but now there's an official guide to "modernize" a setuptools project with first class support in `pyproject.toml`: https://packaging.python.org/en/latest/guides/modernize-setup-py-project/

Cheers,
Kyle

**Comment by lornajane** [comment]
@kyluca yes please, but I think this issue is missing some of the context.

Right now we're using pip-tools with setup.py. I also notice some tox.ini and other files that I don't think are still in use.
We publish the package for others to consume, so our sdist output is also important.

The maintainers on this project aren't Python experts, so we will also need instructions on how to update our existing workflows to use the new approach - we won't already know what to do! If that sounds good, let me know and I'll tag/assign the issue for you.

**Comment by kyluca** [comment]
> The maintainers on this project aren't Python experts, so we will also need instructions on how to update our existing workflows to use the new approach - we won't already know what to do!

No worries!

`pip-tools` is also a great tool (as was `poetry` for its time), although my two-cents on the leading solution for modern projects would be to use `uv`: https://github.com/astral-sh/uv

Even though `uv` is a drop-in replacement for `pip-tools`, it might be worth making this change in two steps to minimise disruption. I can do one PR focused on migrating `setup.py` to `pyproject.toml` while keeping the current usage of `pip-tools`. Then a follow up PR to illustrate how one might adopt `uv` in the place of `pip-tools`. The maintainers can then choose if they're happy to go with the full change or just the first half 👍🏾 

If that's sensible, I'll try to get the first PR up over the next few days 😄

**Comment by lornajane** [comment]
That does sound sensible, thank you!

**Comment by kyluca** [comment]
PR up for the first half, let me know what you think :)

I haven't removed the `tox.ini` yet, just wanted to start with the minimal changes to get `pyproject.toml` fully up and running. Can include the removal (or revival?) of `tox` in the follow up PR.
